### PR TITLE
ttl: default to cluster setting select batch size in ttl job description

### LINKED ISSUE #111208: ttl: default to cluster setting select batch size in ttl job description
If the TTL job defaults to the cluster setting for select batch size it incorrectly shows 0 in the job description (`LIMIT 0`):
```
900024247711662101  ROW LEVEL TTL   "ttl for public.tbl
-- for each span, iterate to find rows:
SELECT signature
FROM public.tbl
AS OF SYSTEM TIME INTERVAL '-30 seconds'
WHERE ((expires_at) <= $1)
AND (
  (signature >= $3)
)
AND (
  (signature <= $2)
)
ORDER BY signature ASC
LIMIT 0
-- then delete with:
DELETE FROM public.tbl
WHERE ((expires_at) <= $1)
AND (signature) IN (($2))"      node    {}  succeeded   NULL    2023-09-15 00:00:02.617121  2023-09-15 00:00:09.411169  2023-09-15 00:00:09.96523   2023-09-15 00:00:09.907439  1   NULL        NULL    1307402813632556058 2023-09-15 00:00:09.411169  2023-09-15 00:00:39.411169  1   {}  []
```

Context:
https://cockroachlabs.slack.com/archives/C05T5HBPYR3/p1695419852757939

Jira issue: CRDB-31821

### PR DESCRIPTION
**Pull request by ecwall**
Fixes #111208

Previously the TTL job description diplayed LIMIT 0 if no select batch size
was specified for the table instead of properly diplaying the cluster setting
value.

This commit changes the code that generates the description to use the same
logic as the job itself which falls back to the cluster setting if no table
storage param is specified.

Release note: None

### CODE REVIEW

File: pkg/sql/ttl/ttlbase/ttl_helpers.go [hunk 1]
```diff
@@ -13,25 +13,90 @@ package ttlbase
...
+		"sql.ttl.default_delete_batch_size",
+		"default amount of rows to delete in a single query during a TTL job",
+		100,
+		settings.PositiveInt,
+		settings.WithPublic,
+	)
+	defaultDeleteRateLimit = settings.RegisterIntSetting(
+		settings.TenantWritable,
+		"sql.ttl.default_delete_rate_limit",
+		"default delete rate limit for all TTL jobs. Use 0 to signify no rate limit.",
```
Comment (chrisseto) [inline pkg/sql/ttl/ttlbase/ttl_helpers.go:53]: nit: can you specify the units on this or further describe how this number is interpreted?

edit: I see now this was just moved. Would be a nice to have to improve the documentation though :D

File: pkg/sql/ttl/ttljob/ttljob.go [hunk 2]
```diff
@@ -40,35 +39,12 @@ import (
...
-		settings.NonNegativeInt,
-		settings.WithPublic)
-
-	jobEnabled = settings.RegisterBoolSetting(
-		settings.TenantWritable,
-		"sql.ttl.job.enabled",
-		"whether the TTL job is enabled",
-		true,
-		settings.WithPublic)
+var jobEnabled = settings.RegisterBoolSetting(
```
Comment (chrisseto) [inline pkg/sql/ttl/ttljob/ttljob.go:42]: Hmm, it feels a bit strange to not see all these settings grouped together. How would you feel about moving this to ttlbase as well? If that's not easy, maybe add a note indicating that settings for the job execution can be found in ttl base?

**Review by chrisseto — APPROVED** [review]
LGTM! Nothing blocking 👍

File: pkg/sql/ttl/ttlbase/ttl_helpers.go (hunk 1 above)
Comment (ecwall) [reply pkg/sql/ttl/ttlbase/ttl_helpers.go:53]: I'll change it now anyways to make it more clear.

File: pkg/sql/ttl/ttljob/ttljob.go (hunk 2 above)
Comment (ecwall) [reply pkg/sql/ttl/ttljob/ttljob.go:42]: I'll move it as well for consistency.

**Comment by ecwall** [comment]
bors r=chrisseto

**Pull request merged**